app/src/main/java/com/epfl/drawyourpath/machineLearning/DigitalInk.kt

```diff
@@ -0,0 +1,61 @@
+package com.epfl.drawyourpath.machineLearning
+
+import android.util.Log
+import com.google.mlkit.common.model.DownloadConditions
+import com.google.mlkit.common.model.RemoteModelManager
+import com.google.mlkit.vision.digitalink.*
+import java.util.concurrent.CompletableFuture
+
+object DigitalInk {
+    /**
+     * Helper function to download an get the ML model
+     * @return the ML model as a future of DigitalInkRecognitionModel
+     */
+    fun downloadModelML(
+        modelIdentifier: DigitalInkRecognitionModelIdentifier = DigitalInkRecognitionModelIdentifier.AUTODRAW,
+    ): CompletableFuture<DigitalInkRecognitionModel> {
+        var result = CompletableFuture<DigitalInkRecognitionModel>()
+
+        var model = DigitalInkRecognitionModel.builder(modelIdentifier).build()
+        val remoteModelManager = RemoteModelManager.getInstance()
+        remoteModelManager.isModelDownloaded(model)
+            .addOnSuccessListener() { is_downloaded: Boolean ->
+                if (!is_downloaded) {
+                    remoteModelManager.download(model, DownloadConditions.Builder().build())
+                        .addOnSuccessListener {
+                            Log.i("Utils", "Model downloaded")

```

Comment:
Forgot to modify the tag.
```suggestion
                            Log.i("DigitalInk", "Model downloaded")

```

Reply:
Thanks, I missed that one !